scripts/ci/gate-registry.py

```
@@ -0,0 +1,208 @@
+            if passthrough and passthrough[0] == "--":
+                passthrough = passthrough[1:]
+            return subprocess.run(
+                [sys.executable, script, "run", *passthrough], cwd=ROOT, check=False
+            ).returncode
+        rendered = command_argv(registry, args.gate)
+        if args.action == "command":
+            print(json.dumps(rendered) if args.json else shlex.join(rendered))
+            return 0
+        if passthrough and passthrough[0] == "--":
+            passthrough = passthrough[1:]
+        return subprocess.run([*rendered, *passthrough], cwd=ROOT, check=False).returncode
```

Ruff (0.16.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)